lib/slog.h

```diff
@@ -24,6 +24,25 @@
 #ifndef SLOG_H_INCLUDED
 #define SLOG_H_INCLUDED 1
 
+#ifdef __APPLE__
+#include <libkern/OSByteOrder.h>
+
+#define htobe16(x) OSSwapHostToBigInt16(x)
```

Comment:
In order to hide missing implementation on platform, or different type of implementation on platforms, there is a so called compat layer in syslog-ng source code. Actually the `compat.h` has a nice description about how to use it: https://github.com/syslog-ng/syslog-ng/blob/3d271976f470832e573528a9eb9c7ca643f9fa67/lib/compat/compat.h#L25-L68

The following defines imho would be better of in a compat layer of their own.

Reply:
BTW could you please explain where exactly are we using these defines ?

Reply:
We have two problems with our code on the MacOS platform.

1. Endianess: On Linux we use le64toh to ensure correct endianess when reading and writing key files. These functions do not exist on MacOS
2. For log verification, we use the hsearch API defined in search.h. However, our implementation currently uses the GNU extensions that also do not exist on MacOS

How would we resolve this using the compatibility mechanism of syslog-ng?

Reply:
Generally the glib is used to handle platform specific differences.
1. Okay, so the disk buffer also has issue with different endianess (you could check that out). There are macros in `glib` that could be useful for you as well: https://developer.gnome.org/glib/stable/glib-Byte-Order-Macros.html
2. Would it be acceptable to replace `hsearch` API with glib hashtable ? That should resolve this issue as well.

Reply:
Thanks for the hints. I will certainly try the glib macros for endianess. Converting to the glib hashtable API is more of an effort. I will look into it when I find time to do so.

Reply:
All checks now pass after I have committed the lasted changes. May I request a review from the core team now?